### PR TITLE
updating README.md samples

### LINKED ISSUE #550: Some code samples are incorrect
Several code samples in the [Readme](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/README.md) have a reference to `tracer.SpanBuilder("incoming request")` but this method does not exist.

I see `SpanBuilderShim : ISpanBuilder` in the Shims.OpenTracing project, but that's the only reference I could find.

**Is this meant to be a future deliverable or is this a mistake?**

Both Zipkin and ApplicationInsights reference this method in the Readme doc, but their sample apps have very different examples.

### PR DESCRIPTION
**Pull request by eddynaka**
Fix #550

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -556,16 +573,17 @@ metricExporter.Start();
...
+var tracer = tracerFactory.GetTracer("application-insights-test");
 
-    await Task.Delay(1000);
-    span.End();
+using (DistributedContext.SetCurrent(dc))
+using (tracer.StartActiveSpan("incoming request", out var span))
+{
+    span.AddEvent("Start processing video.");
+    Thread.Sleep(TimeSpan.FromMilliseconds(10));
+    span.AddEvent("Finished processing video.");
```
Comment (TimothyMothra) [inline README.md]: This sample does not compile. Please give me a few minutes, I'm working on an alternative code sample.

Comment (TimothyMothra) [reply README.md]: @eddynaka Please use this instead:

```
static void Main(string[] args)
{
    DistributedContext.Carrier = AsyncLocalDistributedContextCarrier.Instance; // Enable asynclocal carrier for the context
    DistributedContext dc = DistributedContextBuilder.CreateContext(key: "my.org/keys/frontend", value: "mobile-ios9.3.5");

    using var tracerFactory = TracerFactory.Create(builder => builder
        .SetResource(Resources.CreateServiceResource("my-cloud-role-name"))
        .UseApplicationInsights(config => config.InstrumentationKey = "559ab47f-be09-4558-a0be-3ae747845bca"));

    var tracer = tracerFactory.GetTracer("application-insights-test");

    using (DistributedContext.SetCurrent(dc))
    using (tracer.StartActiveSpan("incoming request", out var span))
    {
        span.AddEvent("Start processing video."); // AI Trace message
        Thread.Sleep(TimeSpan.FromMilliseconds(10));
        span.AddEvent("Finished processing video."); // AI Trace message
    }

    // Note that a manual flush is not necessary. The Application Insights TelemetryClient will be flushed when the tracerFactory is disposed.
}
```

Comment (eddynaka) [reply README.md]: @TimothyMothra , I will update the sample to include the FrontendKey (which is your "my.org/keys/frontend" string). With that, besides the key and the strings that you added, the example is the same.

File: README.md [hunk 2]
```diff
@@ -524,16 +530,27 @@ using (var tracerFactory = TracerFactory.Create(builder => builder
...
-    var tracer = tracerFactory.GetTracer("stackdriver-test");
-    var span = tracer
-        .SpanBuilder("incoming request")
-        .StartSpan();
+var spanExporter = new StackdriverTraceExporter(projectId);
+
+using var tracerFactory = TracerFactory.Create(builder => builder.AddProcessorPipeline(c => c.SetExporter(spanExporter)));
+var tracer = tracerFactory.GetTracer("stackdriver-test");
+
+DistributedContext.Carrier = AsyncLocalDistributedContextCarrier.Instance; // Enable asynclocal carrier for the context
```
Comment (MikeGoldsmith) [inline README.md]: I'm not sure we should be configuring a custom DistributedContext in these examples, this is a separate to the example which is trying to demonstrate the StackDriverExporter.

File: README.md [hunk 3]
```diff
@@ -556,16 +573,21 @@ metricExporter.Start();
...
 ``` csharp
-using (var tracerFactory = TracerFactory.Create(builder => builder
-    .UseApplicationInsights(config => config.InstrumentationKey = "instrumentation-key")))
-{
-    var tracer = tracerFactory.GetTracer("application-insights-test");
-    var span = tracer
-        .SpanBuilder("incoming request")
-        .StartSpan();
+string FrontendKey = "my.org/keys/frontend";
+DistributedContext.Carrier = AsyncLocalDistributedContextCarrier.Instance; // Enable asynclocal carrier for the context
```
Comment (MikeGoldsmith) [inline README.md]: Same as for Traces, I'm not sure we need to introduce / configure a DistributedContext in the example.

Comment (eddynaka) [reply README.md]: should i remove?

File: README.md (hunk 2 above)
Comment (eddynaka) [reply README.md]: should i remove?

Comment (MikeGoldsmith) [reply README.md]: Did you add it for a particular reason? If not, I think we can remove to simplify the examples.

Comment (eddynaka) [reply README.md]: I see. What do you think about this:
```csharp
var spanExporter = new StackdriverTraceExporter(projectId);

using var tracerFactory = TracerFactory.Create(builder => builder.AddProcessorPipeline(c => c.SetExporter(spanExporter)));
var tracer = tracerFactory.GetTracer("stackdriver-test");

using (tracer.StartActiveSpan("/getuser", out TelemetrySpan span))
{
    span.AddEvent("Processing video.");
    span.PutHttpMethodAttribute("GET");
    span.PutHttpHostAttribute("localhost", 8080);
    span.PutHttpPathAttribute("/resource");
    span.PutHttpStatusCodeAttribute(200);
    span.PutHttpUserAgentAttribute("Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0");

    Thread.Sleep(TimeSpan.FromMilliseconds(10));
}
```

File: README.md (hunk 3 above)
Comment (eddynaka) [reply README.md]: What do ou think about this:
```csharp
string FrontendKey = "my.org/keys/frontend";
using var tracerFactory = TracerFactory.Create(builder => builder
    .SetResource(Resources.CreateServiceResource("my-service"))
    .UseApplicationInsights(config => config.InstrumentationKey = "instrumentation-key"));
var tracer = tracerFactory.GetTracer("application-insights-test");

using (tracer.StartActiveSpan("incoming request", out var span))
{
    span.AddEvent("Start processing video.");
    Thread.Sleep(TimeSpan.FromMilliseconds(10));
    span.AddEvent("Finished processing video.");
}
```

**Comment by reyang** [comment]
In general do we want to put all these examples (especially vendor specific exporter related) in the top level README.md file?

Perhaps we can reference this document flow? https://opentelemetry-python.readthedocs.io/en/stable/getting-started.html

**Comment by eddynaka** [comment]
@reyang, thanks for reviewing. Should I change the code to be more simple just like @MikeGoldsmith said?

About the new pr, what are your suggestions? Thank you

File: README.md (hunk 3 above)
Comment (MikeGoldsmith) [reply README.md:577]: Yes, I think this is better. We can clear and concise examples.

**Comment by MikeGoldsmith** [comment]
@reyang yes, I agree we should only demonstrate the API / SDK features of the project and not vendor specific components.

At some point we will push out vendor exporters to their owners org or a contrib repository to tidy up what we have here. We have them here for now to ensure they  continue to compile / work as the raw API & SDK change.

File: README.md (hunk 3 above)
Comment (eddynaka) [reply README.md]: Ok! thanks for reviewing! just updated the samples! :)

**Comment by MikeGoldsmith** [comment]
Thanks @eddynaka.

**Comment by reyang** [comment]
> About the new pr, what are your suggestions? Thank you

I was thinking something similar to https://opentelemetry-python.readthedocs.io/en/stable/getting-started.html. We will need to discuss. I can send a PR suggesting the structure/flow of our docs.

**Pull request merged**